javalib/src/main/scala/java/util/concurrent/ConcurrentHashMap.scala

```diff
@@ -0,0 +1,242 @@
+    inner.equals(o)
+
+  override def putIfAbsent(key: K, value: V): V =
+    inner.putIfAbsent(key, value)
+
+  override def remove(key: Any, value: Any): Boolean =
+    inner.remove(key, value)
+
+  override def replace(key: K, oldValue: V, newValue: V): Boolean =
+    inner.replace(key, oldValue, newValue)
+
+  override def replace(key: K, value: V): V =
+    inner.replace(key, value)
+
+  def contains(value: Any): Boolean =
+    containsValue(value)
+
+  def keys(): Enumeration[K] =
+    Collections.enumeration(inner.keySet())
+
+  def elements(): Enumeration[V] =
+    Collections.enumeration(values())
+}
+
+object ConcurrentHashMap {
+  import HashMap.Node
+
+  /** Inner HashMap that contains the real implementation of a
+   *  ConcurrentHashMap.
+   *
+   *  It is a null-rejecting hash map because some algorithms rely on the fact
+   *  that `get(key) == null` means the key was not in the map.
+   *
+   *  It also has snapshotting iterators to make sure they are *weakly
+   *  consistent*.
+   */
+  private final class InnerHashMap[K, V](
+      initialCapacity: Int,
+      loadFactor: Float
+  ) extends NullRejectingHashMap[K, V](initialCapacity, loadFactor) {
+
+    override private[util] def nodeIterator(): Iterator[HashMap.Node[K, V]] =
+      new NodeIterator
+
+    override private[util] def keyIterator(): Iterator[K] =
+      new KeyIterator
+
+    override private[util] def valueIterator(): Iterator[V] =
+      new ValueIterator
+
+    private def makeSnapshot(): ArrayList[Node[K, V]] = {
+      val snapshot = new ArrayList[Node[K, V]](size())
+      val iter = super.nodeIterator()
+      while (iter.hasNext())
+        snapshot.add(iter.next())
+      snapshot
+    }
+
+    private final class NodeIterator extends AbstractCHMIterator[Node[K, V]] {
+      protected[this] def extract(node: Node[K, V]): Node[K, V] = node
+    }
+
+    private final class KeyIterator extends AbstractCHMIterator[K] {
+      protected[this] def extract(node: Node[K, V]): K = node.key
+    }
+
+    private final class ValueIterator extends AbstractCHMIterator[V] {
+      protected[this] def extract(node: Node[K, V]): V = node.value
+    }
+
+    private abstract class AbstractCHMIterator[A] extends Iterator[A] {
+      private[this] val innerIter = makeSnapshot().iterator()
+      private[this] var lastNode: Node[K, V] = _ // null
+
+      protected[this] def extract(node: Node[K, V]): A
+
+      def hasNext(): Boolean =
+        innerIter.hasNext()
+
+      def next(): A = {
+        val node = innerIter.next()
+        lastNode = node
+        extract(node)
+      }
+
+      override def remove(): Unit = {
+        val last = lastNode
+        if (last eq null)
+          throw new IllegalStateException(
+            "next must be called at least once before remove"
+          )
+        removeNode(last)
+        lastNode = null
+      }
+    }
+  }
+
+  abstract class CollectionView[K, V, E] extends Collection[E]
```

Comment:
I had to add this one to satisfy the linker. Scala.js doesn't need it, and it doesn't appear to be part of the JDK's public API as far as I understand.